### DISCUSSION TITLE
Light file and directory icons shown for dark theme

### DISCUSSION THREAD

**Discussion by pglizniewicz** [doublecmd/doublecmd#1624] [General]
I'm using Double Commander on Fedora 40 with KDE Plasma 6. Is it possible to use "dark" icons for files and directories? To show what I mean, here's a screenshot of Double Commander and Dolphin. Double Commander uses the "light theme icons", which are hardly visible with a dark theme. Dolphin uses the "dark theme icons", I would expect Double Commander to also use these.
![image](https://github.com/doublecmd/doublecmd/assets/4559130/dbd494c2-23db-4cb7-b4f7-69ddb539967b)

**Comment by alexx2000** [comment]
What icon theme do you use?

**Comment by pglizniewicz** [comment]
For Double Commander I'm using the default `dctheme`, it's the only one I see in the dropdown.
For Plasma I'm using the default dark theme: Breeze Dark. It looks like Double Commander is using icons from the light "Breeze" theme.

**Reply by alexx2000** [reply]
If you execute DC from console then you can see which icon theme used.

**Comment by pglizniewicz** [comment]
You mean this?
```
Loading icon theme Breeze Dark
Loading icon theme Breeze
Loading icon theme Hicolor
Loading icon theme DCTheme
```

One odd thing is, the "drive" icons in the "disk panel" are from the Breeze Dark theme, it's only the icons in the "file panel" that are loaded from "light" Breeze.

**Comment by alexx2000** [comment]
Yes, we can see that DC uses `Breeze Dark` icon theme. I cannot reproduce:

![3yjui7j](https://github.com/doublecmd/doublecmd/assets/13401398/9dca8901-dc74-4d1a-a893-9d1808f730b0)

**Reply by alexx2000** [reply]
OK, I found that is wrong. It depends on the icon size.  Folder icon 24x24 (/usr/share/icons/breeze-dark/places/24/folder.svg) has a wrong color:

![folder](https://github.com/doublecmd/doublecmd/assets/13401398/65a0c2dd-8eee-49bf-a312-552fdc1a0ad6)


If you change DC icon size to 16x16 then it should work fine.

**Comment by pglizniewicz** [comment]
Yes, now I see it. I've asked [why is it done this way](https://discuss.kde.org/t/dark-icons-in-dark-breeze/15282), but so far no answer. For some reason, Dolphin uses the 16 px icons when set to 22 px, that's why it looks correct in Dolphin.

**Comment by Kolcha** [comment]
KDE has its own "magic": SVG files in "Breeze" theme are bit special, they are intended to be rendered using KDE frameworks renderer. they has some CSS style built-in (whats definition is available only somewhere internally in renderer), and not always some "fallback" style definition is provided. this is done to change icons color according to the user color settings (notice how folder color in Dolphin changes if you change accent color).

it seems there are few colors in that style definition, including generic text color and background color, which change when you are switching dark/light themes.

if SVG is rendered using KDE frameworks (e.g. Dolphin or other standard KDE app), SVG renderer has access to current color scheme and uses these colors in that CSS style, so icon is rendered properly (that's why Dolphin has 22x22 white icons even they "actually black"). but if the same SVG file is rendered by third-party renderer (i.e. any app able to render SVG) and there is no "fallback style" hardcoded in SVG, it is rendered as black.

here is SVG code for 16x16 folder icon, it has harcoded fallback style (notice inline style definition and `class="ColorScheme-Text"` at the end of `path` element, also `currentColor` "variable" usage in `style` property), so it is (almost) white in any app:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
  <defs id="defs3051">
    <style type="text/css" id="current-color-scheme">.ColorScheme-Text { color: #fcfcfc; } </style>
  </defs>
 <path style="fill:currentColor" d="M 2 2 L 2 3 L 2 6 L 2 7 L 2 13 L 2 14 L 14 14 L 14 13 L 14 6 L 14 5 L 14 4 L 9 4 L 7 2 L 7 2 L 7 2 L 3 2 L 2 2 z M 3 3 L 6.6 3 L 7.6 4 L 7 4 L 7 4 L 7 4 L 5 6 L 3 6 L 3 3 z M 3 7 L 13 7 L 13 13 L 3 13 L 3 7 z " class="ColorScheme-Text"/>
</svg>
```

but what we have for 22x22 folder icon - there is usage of CSS style (`class="ColorScheme-Text"`), but no "generic fallback definition" for it, so the icon is black in any non-KDE app (as no style defined):
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 22 22">
  <defs id="defs3051">
    <style type="text/css" id="current-color-scheme"></style>
  </defs>
    <path style="fill:currentColor" d="M 3 3 L 3 4 L 3 19 L 4 19 L 19 19 L 19 18 L 19 5 L 12 5 L 10 3 L 10 3 L 10 3 L 4 3 L 3 3 z M 4 4 L 7 4 L 9.6 4 L 10.6 5 L 6.6 9 L 6.6 9 L 4 9 L 4 4 z M 9 8 L 18 8 L 18 18 L 4 18 L 4 10 L 5.6 10 L 7 10 L 7 10 L 7 10 L 9 8 z " class="ColorScheme-Text"/>
</svg>
```
there are many icons that don't have "fallback style definition", but not all. maybe some "SVG cleaner" removed them. I'm running Arch Linux, and took this code from my system, but the same was observed previously on Ubuntu 22.04 too. so, this exists at least for a while.

**Reply by alexx2000** [reply]
Yes, I got the same result in my investigation.
